combin/combin.go

```diff
@@ -0,0 +1,169 @@
+		kfact *= i
+	}
+	num := 1
+	for i := n; i > n-k; i-- {
+		num *= i
+	}
+	return num / kfact
+}
+
+// GeneralizedBinomial returns the generalized binomial coefficient of (n, k).
+//  Γ(n+1) / (Γ(k+1) Γ(n-k+1))
+// where Γ is the gamma function. GeneralizedBinomial is useful for continuous
+// relaxations of the binomial coefficient, or when the binomial coefficient value
+// may overflow int. In the latter case, one may alternately use math/big for
+// an exact computation.
+//
+// n and k must be non-negative with n >= k, otherwise GeneralizedBinomial will panic.
+func GeneralizedBinomial(n, k float64) float64 {
+	return math.Exp(LogGeneralizedBinomial(n, k))
+}
+
+// LogGeneralizedBinomial returns the log of the generalized binomial coefficient.
+// See GeneralizedBinomial for more information.
+func LogGeneralizedBinomial(n, k float64) float64 {
+	if n < 0 || k < 0 {
+		panic(badNegInput)
+	}
+	if n < k {
+		panic(badSetSize)
+	}
+	a, _ := math.Lgamma(n + 1)
+	b, _ := math.Lgamma(k + 1)
+	c, _ := math.Lgamma(n - k + 1)
+	return a - b - c
+}
+
+// CombinationGenerator generates combinations iteratively. Combinations may be
+// called to generate all combinations collectively.
+type CombinationGenerator struct {
+	n         int
+	k         int
+	previous  []int
+	remaining int
+}
+
+// NewCombinationGenerator returns a CombinationGenerator for a generating the
+// combinations of k elements from a set of size n.
+//
+// n and k must be non-negative with n >= k, otherwise NewCombinationGenerator
+// will panic.
+func NewCombinationGenerator(n, k int) *CombinationGenerator {
+	combins := Binomial(n, k)
+	return &CombinationGenerator{
+		n:         n,
+		k:         k,
+		remaining: combins,
+	}
+}
+
+// Remaining returns the number of combinations remaining to be generated.
+func (c *CombinationGenerator) Remaining() int {
+	return c.remaining
+}
+
+// Next generates the next combination. If next is non-nil, it must have length
+// k and the result will be stored in-place into next. If next is nil a new
+// slice will be allocated and returned. If all of the combinations have already
+// been constructed (Remaining() returns 0), Next will panic.
+func (c *CombinationGenerator) Next(next []int) []int {
+	if c.remaining == 0 {
+		panic("combin: all combinations have been generated")
+	}
+	if next == nil {
+		next = make([]int, c.k)
+	}
+	if len(next) != c.k {
+		panic(badInput)
+	}
+	if c.previous == nil {
+		c.previous = make([]int, c.k)
+		for i := range c.previous {
+			c.previous[i] = i
+		}
+	} else {
+		nextCombination(c.previous, c.n, c.k)
+	}
+	copy(next, c.previous)
+	c.remaining--
+	return next
+}
+
+// Combinations generates all of the combinations of k elements from a
+// set of size n. The returned slice has length Binomial(n,k) and each inner slice
+// has length k.
+//
+// n and k must be non-negative with n >= k, otherwise Combinations will panic.
+//
+// CombinationGenerator may alternately used to generate each combination
```

Comment:
`s/alternately/alternatively/`.
Also, `be` is missing.
And how about `s/each combination/the combinations/` ?